presto-orc/src/main/java/com/facebook/presto/orc/OrcWriterStats.java

```diff
@@ -13,37 +13,43 @@
  */
 package com.facebook.presto.orc;
 
+import com.facebook.presto.orc.metadata.StripeInformation;
 import org.weakref.jmx.Managed;
 import org.weakref.jmx.Nested;
 
 import java.util.concurrent.atomic.AtomicLong;
 
-import static com.facebook.presto.orc.OrcWriterStats.FlushReason.CLOSED;
-import static com.facebook.presto.orc.OrcWriterStats.FlushReason.DICTIONARY_FULL;
-import static com.facebook.presto.orc.OrcWriterStats.FlushReason.MAX_BYTES;
-import static com.facebook.presto.orc.OrcWriterStats.FlushReason.MAX_ROWS;
+import static com.facebook.presto.orc.WriterStats.FlushReason.CLOSED;
+import static com.facebook.presto.orc.WriterStats.FlushReason.DICTIONARY_FULL;
+import static com.facebook.presto.orc.WriterStats.FlushReason.MAX_BYTES;
+import static com.facebook.presto.orc.WriterStats.FlushReason.MAX_ROWS;
 import static com.google.common.base.MoreObjects.toStringHelper;
 
 public class OrcWriterStats
+        implements WriterStats
 {
-    public enum FlushReason
-    {
-        MAX_ROWS, MAX_BYTES, DICTIONARY_FULL, CLOSED
-    }
-
     private final OrcWriterFlushStats allFlush = new OrcWriterFlushStats("ALL");
     private final OrcWriterFlushStats maxRowsFlush = new OrcWriterFlushStats(MAX_ROWS.name());
     private final OrcWriterFlushStats maxBytesFlush = new OrcWriterFlushStats(MAX_BYTES.name());
     private final OrcWriterFlushStats dictionaryFullFlush = new OrcWriterFlushStats(DICTIONARY_FULL.name());
     private final OrcWriterFlushStats closedFlush = new OrcWriterFlushStats(CLOSED.name());
     private final AtomicLong writerSizeInBytes = new AtomicLong();
 
-    public void recordStripeWritten(FlushReason flushReason, long stripeBytes, int stripeRows, int dictionaryBytes)
+    @Override
+    public void recordStripeWritten(FlushReason flushReason,
+            int stripeMinBytes,
+            int stripeMaxBytes,
+            int dictionaryMaxMemoryBytes,
+            StripeInformation stripeInformation,
```

Comment:
nit: see if it look better if you flip lines 43 and 44 to put the StripeInformation at the end of the method signature.

Reply:
+1

Reply:
I reordered teh parameters. Moved the Config parameters to the front, followed by FlushReason, dictionaryBytes and StripeInformation to the last.